OOPS.cpp

```diff
@@ -0,0 +1,63 @@
+#include <iostream>
+using namespace std;
+
+class Supplier {
+private:
+    int supplierID;
+    const char* supplierName;
+    const char* email;
+    const char* phoneNumber;
+    const char* address;
+
+public:
+    Supplier(int id, const char* name, const char* email, const char* phone, const char* addr)
+        : supplierID(id), supplierName(name), email(email), phoneNumber(phone), address(addr) {}
+
+    int getSupplierID() const { return supplierID; }
+    const char* getSupplierName() const { return supplierName; }
+    const char* getEmail() const { return email; }
+    const char* getPhoneNumber() const { return phoneNumber; }
+    const char* getAddress() const { return address; }
+};
+
+class Warehouse {
+private:
+    int warehouseID;
+    const char* warehouseName;
```

Comment:
Consider using `std::string` instead of `char*`. Although its valid to use it, but `std::string` simplifies memory management and provides better safety.